be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/DataRegistry.scala

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.pop.model.network.method.message.data
+
+import scala.util.Try
+
+import scala.collection.immutable.Map
+import ch.epfl.pop.model.network.method.message.data.ObjectType.ObjectType
+import ch.epfl.pop.model.network.method.message.data.ActionType.ActionType
+
+/**
+  * Class representing a key entry pair to [[DataRegistry]] mapping
+  * @param obj first member of the pair
+  * @param action second member of the pair
+  */
+sealed case class EntryPair(val obj: ObjectType, val action: ActionType)
+
+/**
+  * Class representing Metadata of a specific action and object type
+  * @param schemaValidator
+  * @param buildFromJson
+  */
+sealed case class MetaData(val schemaValidator: String => Try[Unit], val buildFromJson: String => MessageData)
```

Comment:
small thing: you don't need to put "val" for paramters in case classes. They are already val by default

Reply:
:+1:

Reply:
It's necessary to add vals now since I also use the final keyword :)